cartographer/mapping/map_builder_test.cc

```diff
@@ -240,6 +267,148 @@ TEST_F(MapBuilderTest, GlobalSlam3D) {
               0.1 * kTravelDistance);
 }
 
+TEST_F(MapBuilderTest, SaveLoadState) {
+  for (int dimensions : {2, 3}) {
+    if (dimensions == 3) {
+      SetOptionsTo3D();
+    }
+    trajectory_builder_options_.mutable_trajectory_builder_2d_options()
+        ->set_use_imu_data(true);
+    BuildMapBuilder();
+    int trajectory_id = map_builder_->AddTrajectoryBuilder(
+        {kRangeSensorId, kIMUSensorId}, trajectory_builder_options_,
+        GetLocalSlamResultCallback());
+    TrajectoryBuilderInterface* trajectory_builder =
+        map_builder_->GetTrajectoryBuilder(trajectory_id);
+    const auto measurements = test::GenerateFakeRangeMeasurements(
```

Comment:
A bit out of scope: I think it should be 'testing::' not 'test::'. At some point we introduced `test::` and now it looks a bit awkward having both especially in one file.

Reply:
Agree, but this should be a separate PR.